Android 12 crash (SDK Target 31)

Hi!

We have a serious problem that prevents us from releasing new versions to the store. When starting the App after this log its execution dies:

[DEBUG] Pushwoosh: sendAppOpenEndTagMigrate
[DEBUG] Pushwoosh: [[BusinessCase]] trigger welcome-inapp
[DEBUG] Pushwoosh: [BusinessCase] welcome-inapp condition not satisfied
[DEBUG] Pushwoosh: [[BusinessCase]] trigger app-update-message
[DEBUG] Pushwoosh: [BusinessCase] app-update-message condition not satisfied
[ERROR] V8Exception: Exception occurred at /app.js:31: Uncaught TypeError: Cannot read property 'intent' of null

We only did this update in tiApp.xml:

<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="31"/>

That is mandatory currently, Google new rules.

If we set <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="30"/>, all works in device & simulator.

Please, we need urgent help. Any idea? Thank you!

Sure. I've checked TiApp and looks ok.

Scenario 1:

` <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="30"/>
`
ALL GOOD :)

 `<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="31"/>`

Fails! :(

`[INFO]  TiApplication: (main) [0,0] checkpoint, app created.
[INFO]  FirebaseApp: Device unlocked: initializing all Firebase APIs for app [DEFAULT]
[INFO]  FirebaseInitProvider: FirebaseApp initialization successful
[ERROR] Pushwoosh: This is not an Amazon device. The service is not available.
[DEBUG] Pushwoosh: Initializing InternalCrashAnalyticsModule
[DEBUG] Pushwoosh: Current handler class = com.android.internal.os.RuntimeInit$KillApplicationHandler
[DEBUG] Pushwoosh: Executing CrashManager
[DEBUG] Pushwoosh: Looking for stacktraces to send.
[DEBUG] Pushwoosh: Looking for exceptions in: /data/user/0/city.mapyour/files
[DEBUG] Pushwoosh: No stacktraces were found.
[WARN]  ziparchive: Unable to open '/data/user_de/0/com.google.android.gms/app_chimera/m/0000017a/DynamiteLoader.dm': No such file or directory
[WARN]  ziparchive: Unable to open '/data/user_de/0/com.google.android.gms/app_chimera/m/0000017a/DynamiteLoader.dm': No such file or directory
[INFO]  DynamiteModule: Considering local module com.google.android.gms.measurement.dynamite:73 and remote module com.google.android.gms.measurement.dynamite:87
[INFO]  DynamiteModule: Selected remote version of com.google.android.gms.measurement.dynamite, version >= 87
[INFO]  Pushwoosh: Log level: DEBUG
[DEBUG] Pushwoosh: [PushRegistrarHelper] Initializing default PushRegistrar in a plugin
[ERROR] Pushwoosh: This is not an Amazon device. The service is not available.
[WARN]  System: ClassLoader referenced unknown path:
[DEBUG] nativeloader: Configuring classloader-namespace for other apk . target_sdk_version=33, uses_libraries=, library_path=/data/app/~~4vT6ocMfjAgNNk8mdQ8csQ==/com.google.android.gms-k1XwEgLYs2c33htYORENZQ==/lib/arm64:/data/app/~~4vT6ocMfjAgNNk8mdQ8csQ==/com.google.android.gms-k1XwEgLYs2c33htYORENZQ==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.google.android.gms
[DEBUG] Pushwoosh: initHwid
[INFO]  Pushwoosh: [NotificationManager] Sender ID: 1064573959894
[INFO]  Pushwoosh: [NotificationManager] App ID: CDCFE-20872
[DEBUG] Pushwoosh: appOpen:false onAppReady:false
[INFO]  Pushwoosh: HWID: 05494ff6-7228-4685-a9c8-ca6c53380245
[DEBUG] Pushwoosh: [PushwooshModule] onApplicationCreated
[INFO]  Pushwoosh: [d] This is Android FCM device
[WARN]  ziparchive: Unable to open '/data/user_de/0/com.google.android.gms/app_chimera/m/0000017f/MeasurementDynamite.dm': No such file or directory
[WARN]  ziparchive: Unable to open '/data/user_de/0/com.google.android.gms/app_chimera/m/0000017f/MeasurementDynamite.dm': No such file or directory
[DEBUG] CompatibilityChangeReporter: Compat change id reported: 160794467; UID 10010; state: ENABLED
[INFO]  Process: Sending signal. PID: 14599 SIG: 9`

--- a/Module-Source/android/build.gradle
+++ b/Module-Source/android/build.gradle
@@ -1,5 +1,5 @@
 ext {
-    pushwoosh = "6.4.4"
+    pushwoosh = "6.6.8"
 }
 
 dependencies {
@@ -7,5 +7,5 @@ dependencies {
     implementation "com.pushwoosh:pushwoosh-firebase:${pushwoosh}"
     implementation "com.pushwoosh:pushwoosh-amazon:${pushwoosh}"
     implementation "com.pushwoosh:pushwoosh-badge:${pushwoosh}"
-    implementation "com.google.firebase:firebase-messaging:(+,20.0.99]"
+    implementation "com.google.firebase:firebase-messaging:(+,23.1.1]"
 }

--- a/Module-Source/android/build.gradle
+++ b/Module-Source/android/build.gradle
@@ -7,5 +7,5 @@ dependencies {
     implementation "com.pushwoosh:pushwoosh-firebase:${pushwoosh}"
     implementation "com.pushwoosh:pushwoosh-amazon:${pushwoosh}"
     implementation "com.pushwoosh:pushwoosh-badge:${pushwoosh}"
-    implementation "com.google.firebase:firebase-messaging:(+,23.1.1]"
+    implementation "com.google.firebase:firebase-messaging:(+,20.0.99]"
 }

--- a/Module-Source/android/build.gradle
+++ b/Module-Source/android/build.gradle
@@ -1,5 +1,5 @@
 ext {
-    pushwoosh = "6.6.8"
+    pushwoosh = "6.6.9"
 }
 
 dependencies {

--- a/Module-Source/android/manifest
+++ b/Module-Source/android/manifest
@@ -2,7 +2,7 @@
 # this is your module manifest and used by Titanium
 # during compilation, packaging, distribution, etc.
 #
-version: 6.0.9
+version: 6.1.0
 apiversion: 4
 architectures: arm64-v8a armeabi-v7a x86 x86_64
 description: pushnotifications

--- a/Module-Source/android/src/com/pushwoosh/module/PushnotificationsModule.java
+++ b/Module-Source/android/src/com/pushwoosh/module/PushnotificationsModule.java
@@ -26,7 +26,6 @@ import com.pushwoosh.exception.RegisterForPushNotificationsException;
 import com.pushwoosh.function.Callback;
 import com.pushwoosh.function.Result;
 import com.pushwoosh.inapp.InAppManager;
-import com.pushwoosh.inapp.PushwooshInApp;
 import com.pushwoosh.internal.utils.JsonUtils;
 import com.pushwoosh.notification.LocalNotification;
 import com.pushwoosh.notification.LocalNotificationReceiver;

--- a/Module-Source/android/src/com/pushwoosh/module/PushnotificationsModule.java
+++ b/Module-Source/android/src/com/pushwoosh/module/PushnotificationsModule.java
@@ -19,11 +19,13 @@ import android.text.TextUtils;
 import androidx.annotation.NonNull;
 
 import com.pushwoosh.Pushwoosh;
+import com.pushwoosh.RegisterForPushNotificationsResultData;
 import com.pushwoosh.badge.PushwooshBadge;
 import com.pushwoosh.exception.GetTagsException;
 import com.pushwoosh.exception.RegisterForPushNotificationsException;
 import com.pushwoosh.function.Callback;
 import com.pushwoosh.function.Result;
+import com.pushwoosh.inapp.InAppManager;
 import com.pushwoosh.inapp.PushwooshInApp;
 import com.pushwoosh.internal.utils.JsonUtils;
 import com.pushwoosh.notification.LocalNotification;
@@ -149,11 +151,11 @@ public class PushnotificationsModule extends KrollModule {
 		registrationSuccessCallback.set(success);
 		registrationErrorCallback.set(error);
 
-		Pushwoosh.getInstance().registerForPushNotifications(new Callback<String, RegisterForPushNotificationsException>() {
+		Pushwoosh.getInstance().registerForPushNotifications(new Callback<RegisterForPushNotificationsResultData, RegisterForPushNotificationsException>() {
 			@Override
-			public void process(Result<String, RegisterForPushNotificationsException> result) {
+			public void process(@NonNull Result<RegisterForPushNotificationsResultData, RegisterForPushNotificationsException> result) {
 				if (result.isSuccess()) {
-					onRegistrationSucceed(result.getData());
+					onRegistrationSucceed(result.getData().getToken());
 				} else if (result.getException() != null) {
 					onRegistrationFailed(result.getException().getLocalizedMessage());
 				}
@@ -170,7 +172,7 @@ public class PushnotificationsModule extends KrollModule {
 
 	@Kroll.method
 	public void setTags(HashMap params) {
-		Pushwoosh.getInstance().sendTags(Tags.fromJson(JsonUtils.mapToJson(params)), null);
+		Pushwoosh.getInstance().setTags(Tags.fromJson(JsonUtils.mapToJson(params)), null);
 	}
 
 	@Kroll.method
@@ -240,12 +242,12 @@ public class PushnotificationsModule extends KrollModule {
 
 	@Kroll.method
 	public void setUserId(String userId) {
-		PushwooshInApp.getInstance().setUserId(userId);
+		Pushwoosh.getInstance().setUserId(userId);
 	}
 
 	@Kroll.method
 	public void postEvent(String event, HashMap<String, Object> attributes) {
-		PushwooshInApp.getInstance().postEvent(event, Tags.fromJson(JsonUtils.mapToJson((Map<String, Object>) attributes)));
+		InAppManager.getInstance().postEvent(event, Tags.fromJson(JsonUtils.mapToJson((Map<String, Object>) attributes)));
 	}
 
 	@Kroll.method

